crates/context_aware_config/src/api/dimension/types.rs

```diff
@@ -90,3 +90,24 @@ impl DimensionWithMandatory {
         }
     }
 }
+
+#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into)]
+#[serde(try_from = "String")]
+pub struct DeleteReq(String);
+impl DeleteReq {
+    pub fn validate_data(name: String) -> Result<Self, String> {
+        let name = name.trim();
+        if name == "variantIds" {
+            Err("variantIds cannot be deleted".to_string())
+        } else {
+            Ok(Self(name.to_owned()))
+        }
+    }
+}
```

Comment:
Is this used only in try_from? Can this code be there itself? Do we need this impl?